catalog/spatial/wfs/1.0.0/spatial-wfs-v1_0_0-source/src/test/java/org/codice/ddf/spatial/ogc/wfs/catalog/source/reader/TestFeatureCollectionMessageBodyReader.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.wfs.catalog.source.reader;
+
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import javax.ws.rs.WebApplicationException;
+import org.codice.ddf.spatial.ogc.wfs.catalog.common.WfsFeatureCollection;
+import org.codice.ddf.spatial.ogc.wfs.v1_0_0.catalog.source.reader.FeatureCollectionMessageBodyReaderWfs10;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TestFeatureCollectionMessageBodyReader {
+  private static BufferedInputStream validWfsFeatureCollectionResponseXml;
+  private static BufferedInputStream dynamicProxySerializedXml;
+
+  @BeforeClass
+  public static void setUp() {
+    validWfsFeatureCollectionResponseXml =
```

Comment:
❗️ These streams that you're opening here in the the setup method are not being closed properly. Better to provide methods to open the streams that you can call from your test method so that you can properly close them in each test.

Reply:
Got it.